src/StakeRegistry.sol

```diff
@@ -704,4 +712,71 @@ contract StakeRegistry is StakeRegistryStorage {
+     * @dev This function is used by an offchain actor to determine which strategies are restakeable
+     */
+    function getRestakeableStrategies() external view returns (IStrategy[] memory){
+        if(quorumBitmap == 0) {
+            return new IStrategy[](0);
+        }
+
+        bytes memory quorumNumbers = BitmapUtils.bitmapToBytesArray(quorumBitmap);
+        IStrategy[] memory strategies = new IStrategy[](_getNumStrategiesInBitmap(quorumNumbers));
+        uint256 index = 0;
+        for(uint256 i = 0; i < quorumNumbers.length; i++){
+            StrategyParams[] memory strategyParams = strategyParams[uint8(quorumNumbers[i])];
+            for(uint256 j = 0; j < strategyParams.length; j++){
+                strategies[index] = strategyParams[j].strategy;
+                index++;
+            }
+        }
+
+        return strategies;
+    }
+
+    /**
+     * @notice Return a list of all strategies and corresponding share amounts for which the operator has restaked
+     * @dev There may strategies for which the operator has restaked on the quorum but has no shares. In this case,
+     *      the strategy is included in the returned array but the share amount is 0
+     */
+    function getOperatorRestakedStrategies(address operator) external view returns (IStrategy[] memory, uint256[] memory){
+        bytes32 operatorId = registryCoordinator.getOperatorId(operator);
+        uint192 operatorBitmap = registryCoordinator.getCurrentQuorumBitmap(operatorId);
+
+        // Return empty arrays if operator does not have any shares at stake OR no strategies can be restaked
+        if (operatorBitmap == 0 || quorumBitmap == 0) {
+            return (new IStrategy[](0), new uint256[](0));
+        }
+
+        uint192 operatorRestakedQuorumsBitmap = operatorBitmap & quorumBitmap; // get all strategies that are considered restaked
+        bytes memory restakedQuorums = BitmapUtils.bitmapToBytesArray(operatorRestakedQuorumsBitmap);
+        IStrategy[] memory strategies = new IStrategy[](_getNumStrategiesInBitmap(restakedQuorums));
+        uint256[] memory shares = new uint256[](strategies.length);
+
+        uint256 index = 0;
+        for(uint256 i = 0; i < restakedQuorums.length; i++) {
+            StrategyParams[] memory strategyParams = strategyParams[uint8(restakedQuorums[i])];
+            for (uint256 j = 0; j < strategyParams.length; j++) {
+                strategies[index] = strategyParams[j].strategy;
+                shares[index] = delegation.operatorShares(operator, strategyParams[j].strategy);
+                index++;
+            }
+        }
+
+        return (strategies, shares);
+    }
+
+    function _getNumStrategiesInBitmap(bytes memory quorums) internal view returns (uint256) {
+        uint256 strategyCount;
+        for(uint256 i = 0; i < quorums.length; i++) {
+            strategyCount += strategyParams[uint8(quorums[i])].length;
+        }
+        return strategyCount;
+    }
```

Comment:
quick thought:
if the same strategy is shared between any quorums, this will double-count it.
doesn't seem like a config anyone _should_ use, but this just occurred to me

I also think the use of `Bitmap` in this function name is a misnomer, since it's taking a bytes object as input

Reply:
Ya, occurred to me as well. We have double counting where strategies are added twice and "null" counting where strategies are in a quorum and the operator hasn't restated for every strategy in said quorum. The indexing isn't perfect and the backend will end up de-duping and discarding in both cases

Reply:
Ya, naming is stale, will update to bytes

Reply:
OK, seems reasonable, probably want to add a dev tag or some other explanatory note though!
even some like just `@dev does not de-duplicate the list`